primitives/blockchain/src/backend.rs

```diff
@@ -21,9 +21,11 @@ use log::warn;
 use parking_lot::RwLock;
 use sp_runtime::{
 	generic::BlockId,
-	traits::{Block as BlockT, Header as HeaderT, NumberFor},
+	sp_std,
+	traits::{Block as BlockT, Header as HeaderT, NumberFor, Saturating},
 	Justifications,
 };
+use sp_std::collections::btree_set::BTreeSet;
```

Comment:
Nit: iiuc you can use `std` here - backend is not supposed to be compiled for `no_std` env.